src/TwigComponent/src/DependencyInjection/TwigComponentExtension.php

```diff
@@ -31,13 +31,18 @@ final class TwigComponentExtension extends Extension
 {
     public function load(array $configs, ContainerBuilder $container): void
     {
-        $container->registerForAutoconfiguration(ComponentInterface::class)
-            ->addTag('twig.component')
-        ;
+        if (method_exists($container, 'registerAttributeForAutoconfiguration')) {
+            $container->registerAttributeForAutoconfiguration(
+                AsTwigComponent::class,
+                static function (ChildDefinition $definition) {
+                    $definition->addTag('twig.component');
+                }
+            );
+        }
 
         $container->register(ComponentFactory::class)
             ->setArguments([
-                new ServiceLocatorArgument(new TaggedIteratorArgument('twig.component', null, 'getComponentName')),
+                new ServiceLocatorArgument(),
```

Comment:
Use an abstract argument here? Since it will entirely be filled in via the compiler pass?

Reply:
Cool, didn't know about this but `AbstractArgument` is a 5.1+ feature.

Reply:
Yea, I think so. Its presence will remind us that we can remove the conditional code someday :).